snaps/learn/tutorials/transaction-insights.md

```diff
@@ -32,70 +38,493 @@ starter kit by running:
+
+<details>
+  <summary>Did you get a warning?</summary>
+  <div>
+You may get a warning such as: 
 
 ```bash
-npm create @metamask/snap transaction-insights-snap
+@lavamoat/allow-scripts has detected dependencies without configuration. explicit configuration required.
+run "allow-scripts auto" to automatically populate the configuration.
 ```
 
-Next, `cd` into the `transaction-insights-snap` project directory and run:
+You can resolve the issue by running: 
 
 ```bash
-yarn install
-```
+yarn run allow-scripts auto
+``` 
+ </div>
+</details>
 
-This initializes your development environment with the required dependencies.
-You may get a warning similar to the following:
+### 2. (Optional) Customize your Snap's UX
+
+This Snap is generated from a TypeScript template Snap. We recommend customizing your 
+Snap to improve its UX, but this is optional for testing. If you don't wish to customize your Snap, 
+skip to [Step 3](#3-enable-transaction-insights-and-the-ethereum-provider).
+
+ #### 2.1. Provide an icon 
+
+[Optimize your metadata](../best-practices/design-guidelines.md#optimize-your-metadata) and display an 
+icon for your Snap in MetaMask.
+
+Create a new folder `images` in the Snap package `packages/snap/`:
 
 ```bash
-@lavamoat/allow-scripts has detected dependencies without configuration. explicit configuration required.
-run "allow-scripts auto" to automatically populate the configuration.
+mkdir packages/snap/images
 ```
 
-You can resolve the issue by running the following command:
+Download
+[this `gas.svg` icon file](https://raw.githubusercontent.com/Montoya/gas-fee-snap/main/packages/snap/images/gas.svg)
+into that `ìmages` folder.  
+
+<details>
+  <summary>Icon attribution</summary>
+  <div>
+This is a free icon, "Gas" by Mello from the [Noun Project](https://thenounproject.com/browse/icons/term/gas/).
+ </div>
+</details>
+
+
+Your file structure should look like this:
+
+```text
+transaction-insights-snap/
+├─ packages/
+│  ├─ site/
+|  |  |- src/
+|  |  |  |- App.tsx
+|  |  ├─ package.json
+|  |  |- ...(React app content)
+|  |
+│  ├─ snap/
+|  |  ├─ images/
+|  |  |  |- gas.svg
+|  |  ├─ src/
+|  |  |  |- index.test.ts
+|  |  |  |- index.ts
+|  |  ├─ snap.manifest.json
+|  |  ├─ package.json
+|  |  |- ... (Snap content)
+├─ package.json
+├─ ... (other stuff)
+```
 
-```bash
-yarn run allow-scripts auto
+Open `packages/snap/snap.manifest.json` in a text editor. This file contains the main configuration 
+details for your Snap. Edit the `npm` object, within the `location` object, and add `iconPath` with 
+the value `"images/gas.svg"` to point to your new icon:
+
+<Tabs>
+  <TabItem value="Line updates" label="Update snap.manifest.json" default>
+
+```json title="snap.manifest.json"
+"location": {
+  "npm": {
+    "filePath": "dist/bundle.js",
+    "iconPath": "images/gas.svg",
+    "packageName": "snap",
+    "registry": "https://registry.npmjs.org/"
+  }
 ```
+</TabItem>
+<TabItem value="Full page" label="Example" default>
```

Comment:
I don't understand why this example tab is necessary? Can we remove it and make this step the same as in the "Estimate gas" tutorial?